core/dbt/contracts/graph/nodes.py

```diff
@@ -1249,30 +1209,14 @@ def tests(self) -> List[TestDef]:
 
 
 @dataclass
-class ParsedSourceMandatory(GraphNode[GraphResource], HasRelationMetadata):
-    source_name: str
-    source_description: str
-    loader: str
-    identifier: str
-    resource_type: Literal[NodeType.Source]
-
-
-@dataclass
-class SourceDefinition(NodeInfoMixin, ParsedSourceMandatory):
-    quoting: Quoting = field(default_factory=Quoting)
-    loaded_at_field: Optional[str] = None
+class SourceDefinition(
+    NodeInfoMixin,
```

Comment:
Discussed offline: NodeInfoMixin makes a lot of sense to keep as an internal concern of dbt-core, since it is actively set in the internal node + removed prior to serialization. 

We could probably even remove the https://github.com/dbt-labs/dbt-core/pull/9543/files#diff-42cfc99b2fcd89ab75957b8999bee2b1af822f7d39953ca99f1f285c170557c1R1221 method!

The complexity is in ensuring that the SourceDefinition node is not serialized directly via Manifest.writable_manifest, but is first converted to a resource. We'll need to put a better pattern in place to avoid using nodes and resources interchangeably for the purposes of serialization